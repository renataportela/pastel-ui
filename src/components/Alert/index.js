export { default, ALERT_COLORS } from './Alert'