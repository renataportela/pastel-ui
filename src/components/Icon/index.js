export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';