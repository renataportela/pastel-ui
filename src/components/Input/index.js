export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
