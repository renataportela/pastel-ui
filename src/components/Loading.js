import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading