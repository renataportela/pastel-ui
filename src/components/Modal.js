import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;