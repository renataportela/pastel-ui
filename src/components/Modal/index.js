export { default } from './Modal'
export { default as ModalBase } from './ModalBase'