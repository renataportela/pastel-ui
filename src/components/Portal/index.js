export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'