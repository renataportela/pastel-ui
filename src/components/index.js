// Content
export { default as Avatar } from './content/Avatar'
export { default as Badge } from './content/Badge'
export { default as Chip } from './content/Chip'
export { default as Icon, iconNames, Svg } from './content/Icon'
export { default as Tag } from './content/Tag'
export { Heading, Paragraph, Text } from './content/typography'

// Data Entry
export { default as TextField } from './dataEntry/TextField'

// Dialog
export { default as Alert } from './dialog/Alert'

// Layout
export * from './layout/Grid'
export { default as BackTop } from './layout/BackTop'
export { default as Button, Close } from './layout/Button'
export { default as Ripple } from './layout/Ripple'
export { default as Theme, BaseColors, useTheme } from './layout/Theme'