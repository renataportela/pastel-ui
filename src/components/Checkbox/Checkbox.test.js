import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { Theme, Checkbox } from '~/components'

describe('Checkbox component', () => {
  it('renders a unchecked checkbox with label "Option A"', () => {
    const checkbox = shallow(<Theme><Checkbox name="checkbox[]" optionValue="A" label="Option A" value="" onChange={() => {}} /></Theme>)
    expect(shallowToJson(checkbox)).toMatchSnapshot()
  })

  it('renders a checked checkbox with label "Option A"', () => {
    const checkbox = shallow(<Theme><Checkbox name="checkbox[]" optionValue="A" label="Option A" value="A" onChange={() => {}} /></Theme>)
    expect(shallowToJson(checkbox)).toMatchSnapshot()
  })
})