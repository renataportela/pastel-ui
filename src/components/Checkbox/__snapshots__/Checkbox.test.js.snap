// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders a checked checkbox with label "Option A" 1`] = `
<ContextProvider
  value={
    Object {
      "colors": Object {
        "bgColor": "#FFF",
        "danger": Object {
          "bg": "#e24944",
          "hover": "#db332e",
          "text": "#fff",
        },
        "dangerAlt": Object {
          "bg": "#ff7772",
          "hover": "#fc5f5f",
          "text": "#aa1e3a",
        },
        "dark": Object {
          "bg": "#444",
          "hover": "#393939",
          "text": "#fff",
        },
        "darker": Object {
          "bg": "#222",
          "hover": "#202020",
          "text": "#fff",
        },
        "disabled": Object {
          "bg": "#f5f5f5",
          "border": "#ccc",
          "hover": "#e0e0e0",
          "text": "#999",
        },
        "info": Object {
          "bg": "#26e0cd",
          "hover": "#1bd3c1",
          "text": "#fff",
        },
        "infoAlt": Object {
          "bg": "#35ffee",
          "hover": "#2fefdc",
          "text": "#187067",
        },
        "input": Object {
          "bg": "#fcfcfc",
          "border": "#e6e6e6",
          "borderFocused": "#d9aaff",
          "borderHover": "#e8d4f9",
          "disabledBorder": "#f0f0f0",
          "text": "#4a1660",
        },
        "label": "#ad7ed6",
        "neutral": Object {
          "bg": "#FFF",
          "border": "#f0f0f0",
          "hover": "#e8e8e8",
          "text": "#5f5f5f",
        },
        "primary": Object {
          "bg": "#8849bc",
          "hover": "#7b40aa",
          "text": "#fff",
        },
        "primaryAlt": Object {
          "bg": "#d9aaff",
          "hover": "#cc8eff",
          "text": "#4a1660",
        },
        "secondary": Object {
          "bg": "#fc53e5",
          "hover": "#e83ece",
          "text": "#fff",
        },
        "secondaryAlt": Object {
          "bg": "#ff72ee",
          "hover": "#fc53e5",
          "text": "#961284",
        },
        "sub": Object {
          "bg": "#f0f0f0",
          "hover": "#e0e0e0",
          "text": "#444",
        },
        "success": Object {
          "bg": "#96c94a",
          "hover": "#85b53f",
          "text": "#fff",
        },
        "successAlt": Object {
          "bg": "#c7f71b",
          "hover": "#ad0",
          "text": "#465b24",
        },
        "text": "#4a1660",
        "textContrast": "#FFF",
        "transparentDark": "rgba(0, 0, 0, 0.08)",
        "transparentLight": "rgba(255, 255, 255, 0.08)",
        "warning": Object {
          "bg": "#ffcc00",
          "hover": "#ffc700",
          "text": "#fff",
        },
        "warningAlt": Object {
          "bg": "#fff94f",
          "hover": "#fff04f",
          "text": "#ff6e00",
        },
      },
      "fontFamily": "Roboto",
      "fontSize": 16,
    }
  }
>
  <Memo(GlobalStyleComponent)
    theme={
      Object {
        "colors": Object {
          "bgColor": "#FFF",
          "danger": Object {
            "bg": "#e24944",
            "hover": "#db332e",
            "text": "#fff",
          },
          "dangerAlt": Object {
            "bg": "#ff7772",
            "hover": "#fc5f5f",
            "text": "#aa1e3a",
          },
          "dark": Object {
            "bg": "#444",
            "hover": "#393939",
            "text": "#fff",
          },
          "darker": Object {
            "bg": "#222",
            "hover": "#202020",
            "text": "#fff",
          },
          "disabled": Object {
            "bg": "#f5f5f5",
            "border": "#ccc",
            "hover": "#e0e0e0",
            "text": "#999",
          },
          "info": Object {
            "bg": "#26e0cd",
            "hover": "#1bd3c1",
            "text": "#fff",
          },
          "infoAlt": Object {
            "bg": "#35ffee",
            "hover": "#2fefdc",
            "text": "#187067",
          },
          "input": Object {
            "bg": "#fcfcfc",
            "border": "#e6e6e6",
            "borderFocused": "#d9aaff",
            "borderHover": "#e8d4f9",
            "disabledBorder": "#f0f0f0",
            "text": "#4a1660",
          },
          "label": "#ad7ed6",
          "neutral": Object {
            "bg": "#FFF",
            "border": "#f0f0f0",
            "hover": "#e8e8e8",
            "text": "#5f5f5f",
          },
          "primary": Object {
            "bg": "#8849bc",
            "hover": "#7b40aa",
            "text": "#fff",
          },
          "primaryAlt": Object {
            "bg": "#d9aaff",
            "hover": "#cc8eff",
            "text": "#4a1660",
          },
          "secondary": Object {
            "bg": "#fc53e5",
            "hover": "#e83ece",
            "text": "#fff",
          },
          "secondaryAlt": Object {
            "bg": "#ff72ee",
            "hover": "#fc53e5",
            "text": "#961284",
          },
          "sub": Object {
            "bg": "#f0f0f0",
            "hover": "#e0e0e0",
            "text": "#444",
          },
          "success": Object {
            "bg": "#96c94a",
            "hover": "#85b53f",
            "text": "#fff",
          },
          "successAlt": Object {
            "bg": "#c7f71b",
            "hover": "#ad0",
            "text": "#465b24",
          },
          "text": "#4a1660",
          "textContrast": "#FFF",
          "transparentDark": "rgba(0, 0, 0, 0.08)",
          "transparentLight": "rgba(255, 255, 255, 0.08)",
          "warning": Object {
            "bg": "#ffcc00",
            "hover": "#ffc700",
            "text": "#fff",
          },
          "warningAlt": Object {
            "bg": "#fff94f",
            "hover": "#fff04f",
            "text": "#ff6e00",
          },
        },
        "fontFamily": "Roboto",
        "fontSize": 16,
      }
    }
  />
  <Checkbox
    label="Option A"
    name="checkbox[]"
    onChange={[Function]}
    optionValue="A"
    value="A"
  />
</ContextProvider>
`;

exports[`Checkbox component renders a unchecked checkbox with label "Option A" 1`] = `
<ContextProvider
  value={
    Object {
      "colors": Object {
        "bgColor": "#FFF",
        "danger": Object {
          "bg": "#e24944",
          "hover": "#db332e",
          "text": "#fff",
        },
        "dangerAlt": Object {
          "bg": "#ff7772",
          "hover": "#fc5f5f",
          "text": "#aa1e3a",
        },
        "dark": Object {
          "bg": "#444",
          "hover": "#393939",
          "text": "#fff",
        },
        "darker": Object {
          "bg": "#222",
          "hover": "#202020",
          "text": "#fff",
        },
        "disabled": Object {
          "bg": "#f5f5f5",
          "border": "#ccc",
          "hover": "#e0e0e0",
          "text": "#999",
        },
        "info": Object {
          "bg": "#26e0cd",
          "hover": "#1bd3c1",
          "text": "#fff",
        },
        "infoAlt": Object {
          "bg": "#35ffee",
          "hover": "#2fefdc",
          "text": "#187067",
        },
        "input": Object {
          "bg": "#fcfcfc",
          "border": "#e6e6e6",
          "borderFocused": "#d9aaff",
          "borderHover": "#e8d4f9",
          "disabledBorder": "#f0f0f0",
          "text": "#4a1660",
        },
        "label": "#ad7ed6",
        "neutral": Object {
          "bg": "#FFF",
          "border": "#f0f0f0",
          "hover": "#e8e8e8",
          "text": "#5f5f5f",
        },
        "primary": Object {
          "bg": "#8849bc",
          "hover": "#7b40aa",
          "text": "#fff",
        },
        "primaryAlt": Object {
          "bg": "#d9aaff",
          "hover": "#cc8eff",
          "text": "#4a1660",
        },
        "secondary": Object {
          "bg": "#fc53e5",
          "hover": "#e83ece",
          "text": "#fff",
        },
        "secondaryAlt": Object {
          "bg": "#ff72ee",
          "hover": "#fc53e5",
          "text": "#961284",
        },
        "sub": Object {
          "bg": "#f0f0f0",
          "hover": "#e0e0e0",
          "text": "#444",
        },
        "success": Object {
          "bg": "#96c94a",
          "hover": "#85b53f",
          "text": "#fff",
        },
        "successAlt": Object {
          "bg": "#c7f71b",
          "hover": "#ad0",
          "text": "#465b24",
        },
        "text": "#4a1660",
        "textContrast": "#FFF",
        "transparentDark": "rgba(0, 0, 0, 0.08)",
        "transparentLight": "rgba(255, 255, 255, 0.08)",
        "warning": Object {
          "bg": "#ffcc00",
          "hover": "#ffc700",
          "text": "#fff",
        },
        "warningAlt": Object {
          "bg": "#fff94f",
          "hover": "#fff04f",
          "text": "#ff6e00",
        },
      },
      "fontFamily": "Roboto",
      "fontSize": 16,
    }
  }
>
  <Memo(GlobalStyleComponent)
    theme={
      Object {
        "colors": Object {
          "bgColor": "#FFF",
          "danger": Object {
            "bg": "#e24944",
            "hover": "#db332e",
            "text": "#fff",
          },
          "dangerAlt": Object {
            "bg": "#ff7772",
            "hover": "#fc5f5f",
            "text": "#aa1e3a",
          },
          "dark": Object {
            "bg": "#444",
            "hover": "#393939",
            "text": "#fff",
          },
          "darker": Object {
            "bg": "#222",
            "hover": "#202020",
            "text": "#fff",
          },
          "disabled": Object {
            "bg": "#f5f5f5",
            "border": "#ccc",
            "hover": "#e0e0e0",
            "text": "#999",
          },
          "info": Object {
            "bg": "#26e0cd",
            "hover": "#1bd3c1",
            "text": "#fff",
          },
          "infoAlt": Object {
            "bg": "#35ffee",
            "hover": "#2fefdc",
            "text": "#187067",
          },
          "input": Object {
            "bg": "#fcfcfc",
            "border": "#e6e6e6",
            "borderFocused": "#d9aaff",
            "borderHover": "#e8d4f9",
            "disabledBorder": "#f0f0f0",
            "text": "#4a1660",
          },
          "label": "#ad7ed6",
          "neutral": Object {
            "bg": "#FFF",
            "border": "#f0f0f0",
            "hover": "#e8e8e8",
            "text": "#5f5f5f",
          },
          "primary": Object {
            "bg": "#8849bc",
            "hover": "#7b40aa",
            "text": "#fff",
          },
          "primaryAlt": Object {
            "bg": "#d9aaff",
            "hover": "#cc8eff",
            "text": "#4a1660",
          },
          "secondary": Object {
            "bg": "#fc53e5",
            "hover": "#e83ece",
            "text": "#fff",
          },
          "secondaryAlt": Object {
            "bg": "#ff72ee",
            "hover": "#fc53e5",
            "text": "#961284",
          },
          "sub": Object {
            "bg": "#f0f0f0",
            "hover": "#e0e0e0",
            "text": "#444",
          },
          "success": Object {
            "bg": "#96c94a",
            "hover": "#85b53f",
            "text": "#fff",
          },
          "successAlt": Object {
            "bg": "#c7f71b",
            "hover": "#ad0",
            "text": "#465b24",
          },
          "text": "#4a1660",
          "textContrast": "#FFF",
          "transparentDark": "rgba(0, 0, 0, 0.08)",
          "transparentLight": "rgba(255, 255, 255, 0.08)",
          "warning": Object {
            "bg": "#ffcc00",
            "hover": "#ffc700",
            "text": "#fff",
          },
          "warningAlt": Object {
            "bg": "#fff94f",
            "hover": "#fff04f",
            "text": "#ff6e00",
          },
        },
        "fontFamily": "Roboto",
        "fontSize": 16,
      }
    }
  />
  <Checkbox
    label="Option A"
    name="checkbox[]"
    onChange={[Function]}
    optionValue="A"
    value=""
  />
</ContextProvider>
`;
